projects/packages/publicize/src/class-publicize-ui.php

```diff
@@ -592,38 +588,23 @@ private function get_metabox_form_connected( $connections_data ) {
 		<?php
 
 		foreach ( $connections_data as $connection_data ) {
-			$all_connections_done = $all_connections_done && $connection_data['done'];
-			$connection_healthy   = ! isset( $connection_data['is_healthy'] ) || $connection_data['is_healthy'];
-			if ( ! $connection_healthy ) {
-				$broken_connections[] = $connection_data;
-
-			}
 			?>
 
 			<li>
 				<label
-					for="wpas-submit-<?php echo esc_attr( $connection_data['id'] ); ?>"
+					for="wpas-submit-<?php echo esc_attr( $connection_data['connection_id'] ); ?>"
 				>
 					<input
 						type="checkbox"
-						name="wpas[submit][<?php echo esc_attr( $connection_data['id'] ); ?>]"
-						id="wpas-submit-<?php echo esc_attr( $connection_data['id'] ); ?>"
+						name="wpas[submit][<?php echo esc_attr( $connection_data['connection_id'] ); ?>]"
+						id="wpas-submit-<?php echo esc_attr( $connection_data['connection_id'] ); ?>"
 						class="wpas-submit-<?php echo esc_attr( $connection_data['service_name'] ); ?>"
 						value="1"
-						data-id="<?php echo esc_attr( $connection_data['id'] ); ?>"
+						data-id="<?php echo esc_attr( $connection_data['connection_id'] ); ?>"
 					<?php
-						checked( true, $connection_data['enabled'] && $connection_healthy );
-						disabled( false, $connection_healthy );
```

Comment:
This is an interesting flaw currently. If we know that the connection is broken from the last test, then we should update the cache to show that. Currently the cache might have all connections with an `ok` status, but the test-connections call then disabled it.

I think it's probably a follow-up though. It's how it also works in the block editor, but should we remove this disabled logic? Might it be better to update it to check that the status is ok?

Reply:
As mentioned above, the status is by default `null` until connection tests are run.

May be in a separate change, we should use that connection test result call to save the connections list and then we could use the status field from there.

Reply:
Yes, I don't think it's for this change, but we don't actually store the status of the connections now do we? So in the frontend, that's only reflected after the test connections call. I think we should update the connections cache with the result from the test connections call, and that should mean things update and jump around less.

We might want to check we're storing the status of the connections properly on WPCOM too, or we will try and share to broken connections. I think this bit is fine, but something to check while we're at it.

Reply:
Yeah, WPCOM part is going to be tricky.